census/vanilla-tf/trainer/task.py

```diff
@@ -0,0 +1,338 @@
+          job_name, job_id, step, max_steps, accuracy))
+
+  return train_step
+
+
+#
+# Single and Distributed training functions.
+#
+def training_single(job_dir, max_steps, train_eval_tensor):
+  """Perform single node training."""
+
+  inputs, labels = get_placeholders()
+
+  graph = make_graph(inputs, labels)
+  init = tf.global_variables_initializer()
+
+  session = tf.train.MonitoredTrainingSession(checkpoint_dir=job_dir,
+                                              save_checkpoint_secs=20,
+                                              save_summaries_steps=50)
+  session.run(init)
+
+  training_steps(session, graph, inputs, labels, max_steps, train_eval_tensor)
+
+def training_distributed(job_dir, max_steps, train_eval_tensor):
+  """Perform distributed training."""
+
+  # Parse the TF_CONFIG to create cluster spec and job name
+  # We are doing this manually here to demonstrate
+  # See RunConfig abstraction here https://goo.gl/f4BQMo
+  cluster_spec, job_name, task_index = parse_tf_config()
+
+  # Create and start a server
+  server = tf.train.Server(cluster_spec,
+                           job_name=job_name,
+                           task_index=task_index)
+
+  is_chief = (job_name == 'master')
+
+  if job_name == 'ps':
+    server.join()
+  elif job_name in ['master', 'worker']:
+    with tf.device(tf.train.replica_device_setter()):
+      inputs, labels = get_placeholders()
+      graph = make_graph(inputs, labels)
+
+    init = tf.global_variables_initializer()
+
+    with tf.train.MonitoredTrainingSession(master=server.target,
+                                           is_chief=is_chief,
+                                           checkpoint_dir=job_dir,
+                                           save_checkpoint_secs=20,
+                                           save_summaries_steps=50) as session:
+      session.run(init)
+      training_steps(session, graph, inputs, labels, max_steps,
+                     train_eval_tensor, job_name=job_name, job_id=task_index)
+
+
+def evaluate_accuracy(session, eval_step, inputs, labels,
+                      eval_input, eval_label):
+  """Perform the evaluation step to calculate accuracy."""
+  return 100 * session.run(
+      eval_step,
+      feed_dict={
+          inputs: session.run(eval_input),
+          labels: session.run(eval_label)
+      }
+  )
+
+
+def parse_tf_config():
+  """Parse TF_CONFIG to cluster_spec, job_name and task_index."""
+
+  tf_config = os.environ.get('TF_CONFIG')
+
+  if tf_config is None or tf_config == '':
+    return None
+
+  tf_config_json = json.loads(tf_config)
+
+  cluster = tf_config_json.get('cluster')
+  job_name = tf_config_json.get('task').get('type')
+  task_index = tf_config_json.get('task').get('index')
+
+  cluster_spec = tf.train.ClusterSpec(cluster)
+  return cluster_spec, job_name, task_index
+
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser()
```

Comment:
In addition to max_steps, we should have a few other key args -- batch_size, learning_rate, and something like hidden_layer_1, hidden_layer_2 ... so you can show hp tuning, on both the training and the model/graph being built.